File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0
def dfs(numbers, target, idx, values):
    global cnt
    
    if idx == len(numbers):
        if values == target:
            cnt += 1
    else:
        dfs(numbers, target, idx + 1, values + numbers[idx])
        dfs(numbers, target, idx + 1, values - numbers[idx])    

def solution(numbers, target):
    dfs(numbers, target, 0, 0)
    return cnt

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let cnt = 0;
    function dfs(numbers, target, idx, values) {
        if(idx === numbers.length && values === target) {
            cnt += 1;
            return;
        } else if(idx === numbers.length) {
            return;
        }
        dfs(numbers, target, idx + 1, values + numbers[idx]);
        dfs(numbers, target, idx + 1, values - numbers[idx]);
    }
    dfs(numbers, target, 0, 0);
    return cnt;
}